src/state/mod.rs

```diff
@@ -51,7 +51,7 @@ pub struct IndexerState {
     /// needed for the possibility of missing blocks
     pub dangling_branches: Vec<Branch>,
     /// Block database
-    pub indexer_store: Option<IndexerStore>,
+    pub indexer_store: Option<Arc<IndexerStore>>,
```

Comment:
Why would this be optional?

Reply:
It was optional originally, but it definitely shouldn't be.

However, the changes necessary would make this PR noticeably bigger.

Reply:
This `Option` is used for testing, so that we don't need to set up a database connection in order to test nominally pure functions, like `IndexerState::add_block`. Emily's (what's her GH handle?) `MockBlockStore` would solve this issue, although also refactoring the code to better separate effectful and pure code would as well